src/main/java/com/arpnetworking/metrics/mad/parsers/GraphitePlaintextToRecordParser.java

```diff
@@ -0,0 +1,262 @@
+ *     See: http://graphite.readthedocs.io/en/latest/tags.html
+ * 3) Parsing tags from the metric name in InfluxDb format.
+ *     See: https://github.com/influxdata/influxdb/issues/2996
+ *
+ * Sample MAD configuration:
+ * <pre>
+ * {
+ *   type="com.arpnetworking.metrics.mad.sources.MappingSource"
+ *   name="graphitetcp_mapping_source"
+ *   actorName="graphite-tcp-source"
+ *   findAndReplace={
+ *     "\\."=["/"]
+ *   }
+ *   source={
+ *     type="com.arpnetworking.metrics.common.sources.TcpLineSource"
+ *     name="graphitetcp_source"
+ *     host="0.0.0.0"
+ *     port="2003"
+ *   }
+ * }
+ * </pre>
+ *
+ * TODO(ville): Add pickle support.
+ * See:
+ * - https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-pickle-protocol
+ * - https://github.com/irmen/Pyrolite
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public final class GraphitePlaintextToRecordParser implements Parser<List<Record>, ByteBuffer> {
+
+    /**
+     * Parses a graphite plaintext datagram.
+     *
+     * @param datagram a datagram
+     * @return A list of {@link DefaultRecord.Builder}
+     * @throws ParsingException if the datagram is not parsable as graphite plaintext formatted message
+     */
+    public List<Record> parse(final ByteBuffer datagram) throws ParsingException {
+        // CHECKSTYLE.OFF: IllegalInstantiation - This is the recommended way
+        final String datagramAsString = new String(datagram.array(), Charsets.UTF_8);
+        final ImmutableList.Builder<Record> recordListBuilder = ImmutableList.builder();
+        final DateTime now = DateTime.now();
+        for (final String line : LINE_SPLITTER.split(datagramAsString)) {
+            // CHECKSTYLE.ON: IllegalInstantiation
+            final Matcher matcher = GRAPHITE_PATTERN.matcher(line);
+            if (!matcher.matches()) {
+                throw new ParsingException("Invalid graphite line", line.getBytes(Charsets.UTF_8));
+            }
+
+            // Annotations
+            final Map<String, String> dimensions = Maps.newHashMap();
+
+            // Add global dimensions (lowest priority)
+            // IMPORTANT: Subsequently processed tags can overwrite these
+            dimensions.putAll(_globalTags);
+
+            // Parse the name
+            String name = parseName(datagram, matcher.group("NAME"));
+            if (_parseCarbonTags) {
+                // IMPORTANT: Subsequently processed tags can overwrite these
+                final String[] parts = name.split(";");
+                name = parts[0];
+                for (int i = 1; i < parts.length; ++i) {
+                    final String[] subparts = parts[i].split("=");
+                    if (subparts.length == 2 && !subparts[0].isEmpty() && !subparts[1].isEmpty()) {
+                        dimensions.put(subparts[0], subparts[1]);
+                    }
+                    // TODO(ville): should we rate limit log that we potentially dropped something?
```

Comment:
I don't think we should fail the whole bytebuffer for invalid data.  If we want to keep the parser processing multiple records, I think it should discard invalid lines, but still process valid lines.

Reply:
I agree. However, this is a large discussion imo and affects more sinks than just this one. I would like to discuss with you and make changes separately. The goal here was to raise the issue while achieving parity with the current design.